tests/codegen/simd/issue-120720-reduce-nan.rs

```diff
@@ -0,0 +1,17 @@
+//@ compile-flags: -C opt-level=3 -C target-cpu=cannonlake
```

Comment:
I would add a brief (e.g. one line) comment explaining what this test is testing for, to save people having to refer to this PR if they want to know the basics.